apps/aecore/lib/aecore/sync/chain.ex

```diff
@@ -0,0 +1,148 @@
+defmodule Aecore.Sync.Chain do
+  @moduledoc """
+  Implements all the functions regarding the Chain structure of the SyncTask
+  """
+
+  alias Aecore.Chain.Header
+  alias Aecore.Chain.BlockValidation
+  alias Aecore.Sync.Task
+  alias __MODULE__
+
+  @type peer_id :: pid()
+  @type chain_id :: reference()
```

Comment:
what is `reference`?

Reply:
[Basic elixir types](https://hexdocs.pm/elixir/typespecs.html#basic-types)

Reply:
the link does not explain this well

Reply:
This is used as unique identifier. Elixir has such type, named reference(), and we could simply make one using `Kernel.make_ref()`.

Reply:
what does data inside look

Reply:
The data is it's own type - reference -> `#Reference<0.53780266.3535798273.175449>`

Reply:
Okay, I'll have to do more research myself